File: rerun-reports/lane-a/010/20260827-232948/results/event-hubs/data-plane/java/streaming/event-hubs-dp-java-streaming/java-azure-skills/azure-skill-mcp-microsoft-skill/generated-code/src/main/java/com/example/EventHubsDemo.java
package com.example;

import com.azure.messaging.eventhubs.EventData;
import com.azure.messaging.eventhubs.EventDataBatch;
import com.azure.messaging.eventhubs.EventHubClientBuilder;
import com.azure.messaging.eventhubs.EventHubProducerClient;
import com.azure.messaging.eventhubs.EventProcessorClient;
import com.azure.messaging.eventhubs.EventProcessorClientBuilder;
import com.azure.messaging.eventhubs.checkpointstore.blob.BlobCheckpointStore;
import com.azure.messaging.eventhubs.models.ErrorContext;
import com.azure.messaging.eventhubs.models.EventContext;
import com.azure.messaging.eventhubs.models.EventPosition;
import com.azure.storage.blob.BlobContainerAsyncClient;
import com.azure.storage.blob.BlobContainerClientBuilder;

import java.time.Duration;
import java.util.UUID;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;
import java.util.function.Consumer;

public final class EventHubsDemo {
    private static final int EVENT_COUNT = 10;
    private static final Duration RECEIVE_TIMEOUT = Duration.ofSeconds(60);

    private EventHubsDemo() {
    }

    public static void main(String[] args) throws InterruptedException {
        String eventHubsConnectionString = requireEnvironmentVariable(
            "EVENT_HUBS_CONNECTION_STRING");
        String eventHubName = requireEnvironmentVariable("EVENT_HUB_NAME");
        String storageConnectionString = requireEnvironmentVariable(
            "AZURE_STORAGE_CONNECTION_STRING");
        String checkpointContainerName = requireEnvironmentVariable(
            "BLOB_CHECKPOINT_CONTAINER");
        String runId = UUID.randomUUID().toString();

        sendEvents(eventHubsConnectionString, eventHubName, runId);

        BlobContainerAsyncClient blobContainerClient =
            new BlobContainerClientBuilder()
                .connectionString(storageConnectionString)
                .containerName(checkpointContainerName)
                .buildAsyncClient();

        CountDownLatch receivedEvents = new CountDownLatch(EVENT_COUNT);

        Consumer<EventContext> processEvent = eventContext -> {
            EventData event = eventContext.getEventData();
            System.out.printf(
                "Received partition=%s, sequence=%d, body=%s, properties=%s%n",
                eventContext.getPartitionContext().getPartitionId(),
                event.getSequenceNumber(),
                event.getBodyAsString(),
                event.getProperties());

            if (runId.equals(event.getProperties().get("runId"))) {
                receivedEvents.countDown();
            }

            eventContext.updateCheckpoint();
        };

        Consumer<ErrorContext> processError = errorContext -> {
            String partitionId = errorContext.getPartitionContext() == null
                ? "N/A"
                : errorContext.getPartitionContext().getPartitionId();
            System.err.printf(
                "Error while processing partition %s: %s%n",
                partitionId,
                errorContext.getThrowable());
        };

        EventProcessorClient processor = new EventProcessorClientBuilder()
            .connectionString(eventHubsConnectionString, eventHubName)
            .consumerGroup(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME)
            .checkpointStore(new BlobCheckpointStore(blobContainerClient))
            .initialPartitionEventPosition(partitionId -> EventPosition.earliest())
            .processEvent(processEvent)
            .processError(processError)
            .buildEventProcessorClient();

        try {
            processor.start();
            System.out.printf(
                "Processor started; waiting up to %d seconds for this run's events.%n",
                RECEIVE_TIMEOUT.toSeconds());

            if (!receivedEvents.await(RECEIVE_TIMEOUT.toSeconds(), TimeUnit.SECONDS)) {
                throw new IllegalStateException(
                    "Timed out waiting for all events; remaining=" + receivedEvents.getCount());
            }
        } finally {
            processor.stop();
            System.out.println("Processor stopped.");
        }
    }

    private static void sendEvents(
        String connectionString,
        String eventHubName,
        String runId) {

        try (EventHubProducerClient producer = new EventHubClientBuilder()
            .connectionString(connectionString, eventHubName)
            .buildProducerClient()) {

            EventDataBatch batch = producer.createBatch();

            for (int i = 1; i <= EVENT_COUNT; i++) {
                EventData event = new EventData("Demo event " + i);
                event.getProperties().put("eventNumber", i);
                event.getProperties().put("source", "java-event-hubs-demo");
                event.getProperties().put("runId", runId);

                if (!batch.tryAdd(event)) {
                    throw new IllegalStateException(
                        "The 10 demo events do not fit in one EventDataBatch.");
                }
            }

            producer.send(batch);
            System.out.printf("Sent %d events with runId=%s%n", batch.getCount(), runId);
        }
    }

    private static String requireEnvironmentVariable(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException(
                "Required environment variable is not set: " + name);
        }
        return value;
    }
}

File: rerun-reports/lane-b/051/20260828-035813/results/event-hubs/data-plane/python/streaming/event-hubs-dp-python-streaming/python-azure-skills/azure-skill-mcp-microsoft-skill/generated-code/event_hubs_demo.py
"""Send and receive Azure Event Hubs events with Blob Storage checkpoints."""

import asyncio
import os

from azure.eventhub import EventData
from azure.eventhub.aio import EventHubConsumerClient, EventHubProducerClient
from azure.eventhub.extensions.checkpointstoreblobaio import BlobCheckpointStore
from azure.identity.aio import DefaultAzureCredential


def required_setting(name: str) -> str:
    value = os.getenv(name)
    if not value:
        raise RuntimeError(f"Set the {name} environment variable.")
    return value


async def send_events(
    namespace: str,
    event_hub_name: str,
    credential: DefaultAzureCredential,
) -> None:
    async with EventHubProducerClient(
        fully_qualified_namespace=namespace,
        eventhub_name=event_hub_name,
        credential=credential,
    ) as producer:
        batch = await producer.create_batch()
        for event_number in range(1, 6):
            batch.add(EventData(f"Sample event {event_number}"))

        await producer.send_batch(batch)
        print(f"Sent {len(batch)} events.")


async def receive_events(
    namespace: str,
    event_hub_name: str,
    consumer_group: str,
    storage_account_url: str,
    checkpoint_container: str,
    credential: DefaultAzureCredential,
) -> None:
    checkpoint_store = BlobCheckpointStore(
        blob_account_url=storage_account_url,
        container_name=checkpoint_container,
        credential=credential,
    )

    async def on_event(partition_context, event) -> None:
        print(
            f"Partition {partition_context.partition_id}: "
            f"{event.body_as_str(encoding='UTF-8')}"
        )
        await partition_context.update_checkpoint(event)

    async def on_error(partition_context, error: Exception) -> None:
        if partition_context is None:
            print(f"Consumer error: {error}")
        else:
            print(
                f"Consumer error on partition "
                f"{partition_context.partition_id}: {error}"
            )

    async with EventHubConsumerClient(
        fully_qualified_namespace=namespace,
        eventhub_name=event_hub_name,
        consumer_group=consumer_group,
        credential=credential,
        checkpoint_store=checkpoint_store,
    ) as consumer:
        print("Receiving events. Press Ctrl+C to stop.")
        await consumer.receive(
            on_event=on_event,
            on_error=on_error,
            starting_position="-1",
        )


async def main() -> None:
    namespace = required_setting("EVENT_HUB_FULLY_QUALIFIED_NAMESPACE")
    event_hub_name = required_setting("EVENT_HUB_NAME")
    storage_account_url = required_setting("STORAGE_ACCOUNT_URL")
    checkpoint_container = required_setting("CHECKPOINT_CONTAINER")
    consumer_group = os.getenv("EVENT_HUB_CONSUMER_GROUP", "$Default")

    async with DefaultAzureCredential() as credential:
        await send_events(namespace, event_hub_name, credential)
        await receive_events(
            namespace,
            event_hub_name,
            consumer_group,
            storage_account_url,
            checkpoint_container,
            credential,
        )


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        print("Stopped.")
